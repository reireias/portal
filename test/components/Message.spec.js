import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import tartget from '@/components/Message.vue'

const localVue = createLocalVue()

Vue.use(Vuetify)
localVue.use(Vuex)
localVue.use(Vuetify)

describe('Message', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const wrapper = shallowMount(tartget, {
      propsData: {
        message: {
          text: 'text',
          client: 'client',
          createdAt: {
            toDate: () => {
              return new Date('2020/01/01')
            },
          },
        },
      },
      localVue,
      vuetify,
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
