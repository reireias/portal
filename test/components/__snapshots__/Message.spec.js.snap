// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message is a Vue instance 1`] = `
"<v-card-stub loaderheight=\\"4\\" tag=\\"div\\" class=\\"message\\">
  <v-container-stub tag=\\"div\\">
    <v-row-stub tag=\\"div\\">
      <v-col-stub cols=\\"12\\" sm=\\"6\\" tag=\\"div\\">
        text
      </v-col-stub>
      <v-col-stub cols=\\"12\\" sm=\\"3\\" tag=\\"div\\">
        client
      </v-col-stub>
      <v-col-stub cols=\\"12\\" sm=\\"3\\" tag=\\"div\\">
        2020/01/01 00:00:00
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</v-card-stub>"
`;
